DAQ/src/StrawDigisFromArtdaqFragments_module.cc

```diff
@@ -0,0 +1,437 @@
+// ======================================================================
+// PM
+// StrawDigisFromArtdaqFragments:  add tracker data products to the event
+// each diagnostic printout has a level >= 0
+// b) the higher it is, the less important is the printout
+// c) in a job with diagLevel_ set, only printouts with level <= diagLevel_ are enabled
+//
+// ======================================================================
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "fhiclcpp/ParameterSet.h"
+
+#include "art/Framework/Principal/Handle.h"
+#include "artdaq-core-mu2e/Data/TrackerDataDecoder.hh"
+#include "artdaq-core-mu2e/Overlays/FragmentType.hh"
+#include "artdaq-core-mu2e/Overlays/DTC_Packets/DTC_RocDataHeaderPacket.h"
+
+#include "Offline/DataProducts/inc/TrkTypes.hh"
+#include "Offline/RecoDataProducts/inc/IntensityInfoTrackerHits.hh"
+// #include "Offline/RecoDataProducts/inc/ProtonBunchTime.hh"
+#include "Offline/RecoDataProducts/inc/StrawDigi.hh"
+
+#include <artdaq-core/Data/Fragment.hh>
+#include <artdaq-core/Data/ContainerFragment.hh>
+
+#include <iostream>
+
+#include <string>
+
+#include <memory>
+
+// #define TRACEMF_USE_VERBATIM 1
+
+// #include "TRACE/tracemf.h"
+// #define TRACE_NAME "StrawDigisFromArtdaqFragments"
+
+
+namespace art {
+  class StrawDigisFromArtdaqFragments;
+}
+// ======================================================================
+
+class art::StrawDigisFromArtdaqFragments : public EDProducer {
+
+public:
+
+  struct Config {
+    fhicl::Atom<int> diagLevel    {fhicl::Name("diagLevel"    ), fhicl::Comment("diagnostic severity level, default = 0" ), 0};
+    fhicl::Atom<int> debugLevel   {fhicl::Name("debugLevel"   ), fhicl::Comment("debug level, default = 0"               ), 0};
+    fhicl::Atom<int> saveWaveforms{fhicl::Name("saveWaveforms"), fhicl::Comment("save StrawDigiADCWaveforms, default = 1"), 1};
```

Comment:
Is this an option that might be changed in normal production? If so, it should not have a hard-coded default.

Reply:
Also, using 'int' as 'bool' is deprecated. Please change to bool so that values 'true' and 'false' can be set in fcl.